### ISSUE TITLE
Properties of an ES2015 inner class get incorrect longnames

### ISSUE THREAD

**Issue by kruncher** [jsdoc3/jsdoc#1118]
I see that jsdoc has support for ES2015 classes but the documentation doesn't show an example for the following situation:

```
/** Class representing a point. */
class Point {
    constructor(x, y) {
        /**
         * The x value.
         * @type {number}
         */
        this.x = x;
        /**
         * The y value.
         * @member {number}
         */
        this.y = y;
    }

    /**
     * The z value.
     * @type {number} z
     */

    /**
     * The w value.
     * @member {number} w
     */
}
```

In the above scenario properties `x`, `y`, `z` and `w` do not seem to be included in the generated documentation.

What is the proper way to achieve this?

**Comment by hegemonic** [comment]
The following example works perfectly for me in JSDoc 3.4.0:

```js
/** Class representing a point. */
class Point {
    constructor(x, y) {
        /**
         * The x value.
         * @type {number}
         */
        this.x = x;
        /**
         * The y value.
         * @type {number}
         */
        this.y = y;
    }

    /**
     * The z value.
     * @name Point#z
     * @type {number}
     */

    /**
     * The w value.
     * @name Point#w
     * @type {number}
     */
}
```

**Comment by Fr0sT-Brutal** [comment]
Weird... I'm running this sample as is with all versions from 3.4.0 to 3.4.4 with latest Node as well as with LTS and still not getting members.
![image](https://cloud.githubusercontent.com/assets/1469388/22653366/2b7e42f4-ec9a-11e6-8c55-72f91b87c23e.png)
```
/**
  Test
  @module Test
 */


/** Class representing a point. */
class Point {
    constructor(x, y) {
        /**
         * The x value.
         * @type {number}
         */
        this.x = x;
        /**
         * The y value.
         * @type {number}
         */
        this.y = y;
    }

    /**
     * The z value.
     * @name Point#z
     * @type {number}
     */

    /**
     * The w value.
     * @name Point#w
     * @type {number}
     */
}
```

**Comment by hegemonic** [comment]
@Fr0sT-Brutal, because `Point` is an inner member of `module:Test`, your example should actually look like this:

```js
/**
  Test
  @module Test
 */

/** Class representing a point. */
class Point {
    constructor(x, y) {
        /**
         * The x value.
         * @type {number}
         */
        this.x = x;
        /**
         * The y value.
         * @type {number}
         */
        this.y = y;
    }

    /**
     * The z value.
     * @name module:Test~Point#z
     * @type {number}
     */

    /**
     * The w value.
     * @name module:Test~Point#w
     * @type {number}
     */
}
```

But with that updated code, we fail to pick up the `x` and `y` properties—each of them gets an incorrect longname. For that reason, I'm reopening this issue.

As a workaround, you can use the `@alias` tag to force each member to get the correct longname:

```js
/**
  Test
  @module Test
 */

/** Class representing a point. */
class Point {
    constructor(x, y) {
        /**
         * The x value.
         * @alias module:Test~Point#x
         * @type {number}
         */
        this.x = x;
        /**
         * The y value.
         * @alias module:Test~Point#y
         * @type {number}
         */
        this.y = y;
    }

    /**
     * The z value.
     * @name module:Test~Point#z
     * @type {number}
     */

    /**
     * The w value.
     * @name module:Test~Point#w
     * @type {number}
     */
}
```

**Comment by hegemonic** [comment]
This is probably a duplicate of #1182. Need to investigate further.

**Comment by hegemonic** [comment]
Yep, this is a duplicate of #1182, which is now fixed on master. The fix will be included in JSDoc 3.5.0.
